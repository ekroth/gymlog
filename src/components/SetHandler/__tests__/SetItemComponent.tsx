import React from 'react'
import { shallow } from 'enzyme'
import { SetItemComponent, SetItemProps } from '../SetItemComponent'

describe('SetItemComponent', () => {
  describe('rendering', () => {
    it('should match snapshot', () => {
      const props: SetItemProps = {
        index: 0,
        onPress: () => {
          return
        },
        selected: false,
        set: { weight: 150, reps: 5 }
      }
      const wrapper = shallow(<SetItemComponent {...props} />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
