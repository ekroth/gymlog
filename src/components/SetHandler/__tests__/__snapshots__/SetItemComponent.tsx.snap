// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetItemComponent rendering should render a <View /> 1`] = `
<Styled(ListItem)
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "white",
    }
  }
>
  <GridNB>
    <ColumnNB
      size={20}
    >
      <Styled(Text)
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        0
      </Styled(Text)>
    </ColumnNB>
    <ColumnNB
      size={60}
    >
      <GridNB>
        <ColumnNB>
          <Styled(Text)
            style={
              Object {
                "alignSelf": "flex-end",
                "fontWeight": "bold",
              }
            }
          >
            150
          </Styled(Text)>
        </ColumnNB>
        <ColumnNB>
          <Styled(Subtitle)
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
          >
            kgs
          </Styled(Subtitle)>
        </ColumnNB>
      </GridNB>
    </ColumnNB>
    <ColumnNB
      size={10}
    >
      <Styled(Text)
        style={
          Object {
            "alignSelf": "flex-end",
            "fontWeight": "bold",
          }
        }
      >
        5
      </Styled(Text)>
    </ColumnNB>
    <ColumnNB
      size={10}
    >
      <Styled(Subtitle)
        style={
          Object {
            "alignSelf": "flex-start",
          }
        }
      >
        reps
      </Styled(Subtitle)>
    </ColumnNB>
  </GridNB>
</Styled(ListItem)>
`;
